src/main/scala/org/broadinstitute/hail/Utils.scala

```diff
@@ -327,6 +327,49 @@ class RichSparkContext(val sc: SparkContext) extends AnyVal {
       .map(l => Line(l, None, file))
 }
 
+class HeadRDD[T](prev: RDD[T], num: Int, exact: Boolean)(implicit tct: ClassTag[T]) extends RDD[T](prev) {
+
+  private var nLastPart = 0
+  private var lastPart = 0
+
+  def getPartitions: Array[Partition] = {
+    val buf = new mutable.ArrayBuffer[Partition]
+
+    var nvp = prev.sparkContext.runJob(prev,(it: Iterator[T]) => it.size, Seq(0))
+    if(exact) {
+      while (nvp.sum < num && nvp.size < prev.partitions.length) {
+        nvp = nvp ++ prev.sparkContext.runJob(prev, (it: Iterator[T]) => it.size,
+          Range(nvp.length,
+            nvp.length +
+              math.min(((num - nvp.sum) / (nvp.sum / nvp.length).toDouble).ceil.toInt, prev.partitions.length)))
+      }
+
+      def f(n: Int, i: Int): Unit = {
+        if (n + nvp(i) > num) {
+          lastPart = i
+          nLastPart = num - n
+        } else {
+          f(n + nvp(i), i + 1)
+        }
+      }
+      f(0, 0)
+    }else{
+      lastPart = math.min((num / nvp.sum.toDouble).ceil.toInt, prev.partitions.length)
+      nLastPart = Int.MaxValue
+    }
+    firstParent.partitions.take(lastPart+1)
+  }
+  
+  def compute(split: Partition, context: TaskContext): Iterator[T] = {
+
+    if(split.index == lastPart ){
+      firstParent.iterator(split,context).take(nLastPart)
+    }else{
+      firstParent.iterator(split,context)
+    }
```

Comment:
Try to make the formatting consistent with the surrounding code.  I'm kind of anal about code style.  For example, there's an extra space in the if condition, and we normally put spaces around curly braces.  IntelliJ auto-reformat is your friend here: Command-Shift-L.

Also, I prefer not to use curly braces when if arms are a single expression.  (I will add this to the budding style guide: https://github.com/broadinstitute/hail/blob/master/style-guide.md)